internal/rpmmd/repository.go

```diff
@@ -203,6 +203,23 @@ func (re *RepositoryError) Error() string {
 	return re.msg
 }
 
+func GetVerStrFromPackageSpecList(pkgs []PackageSpec, packageName string) (string, error) {
```

Comment:
what about:

```golang
func (pkgs []PackageSpec) FindByName(name string) *PackageSpace {
    for _, pkg := range pkgs {
        if pkg.Name == packageName {
	    return &pkg
	}
    }
    return nil
}

func (p PackageSpec) NRA() string {
    return fmt.Sprintf("%s-%s.%s", p.Version, p.Release, p.Arch)
}

```

The following check:
```golang
p := pkgs.FindByName("osbuild")
if p == nil {
    panic("osbuild is not there, wow")
}
```
might be even omitted because it will panic nevertheless. :shrug: Or maybe we should return an error instead.

Reply:
`FindByNameOrPanic` might be another solution.

Reply:
Thanks for the review.

Unfortunately the first part of your suggestion is not a valid Go code, since `[]PackageSpec` can not be used as a receiver type. I would have to define a type alias for it. I considered this, but decided to not do it as it would require changing the type in too many places. So I decided to define it as a function. However I like the idea of defining `VRA()` as `PackageSpec` method... 

WRT checking the error and panic. It felt too much to add gazillion of `if` statements to check returned error into every place where the function which previously panicked used to be called. Also it felt wrong to me to not provide function variant which returns `error` instead of always panicking. So I decided to implement a wrapper which checks the error and panics if there is some.

I'm happy to rework this a bit based on your suggestion.

Reply:
> `[]PackageSpec` can not be used as a receiver type.

ofc :facepalm: sorry! Then we would have to go with a method similar to yours. Well, whatever works for you, I need to study more Go apparently. :shrug: